### ISSUE TITLE
Java Serialization Handling

### ISSUE THREAD

**Issue by JStyle21** [zaproxy/zaproxy#4509]
Opening this issue per Simon's suggestion regarding my discussion regarding some kind of add on \ plugin to handle java serialization in Zap.

**Comment by psiinon** [comment]
Thanks @JStyle21 - I've just sent you an invite to colaberate on this project, let me know when you've accepted it and we can assign this issue to you :)

**Comment by JStyle21** [comment]
@psiinon Accepted

**Comment by thc202** [comment]
Assigned :)

**Comment by thc202** [comment]
Let us know if you have any questions (e.g. where to start).

**Comment by psiinon** [comment]
Yeah - its a big code base, so dont hesitate to ask for pointers...

**Comment by kingthorin** [comment]
Ref: https://groups.google.com/forum/#!topic/zaproxy-users/QQgE6JqLiRM

If I understand correctly this is to be a new request/response view addon? 
In which case you might want to look at:
https://github.com/zaproxy/zap-extensions/tree/alpha/src/org/zaproxy/zap/extension/jsonview
https://github.com/zaproxy/zap-extensions/tree/alpha/src/org/zaproxy/zap/extension/viewstate

**Comment by thc202** [comment]
That and a `Variant`?

**Comment by JStyle21** [comment]
I was thinking something that not only gives you the ability to view
req\res but also modify and re-serialize from there one can also make some
scripts for automation.

Appreciate the help so far and any help in general.

Thanks :)

**Comment by thc202** [comment]
The views normally allow to modify the contents shown (e.g. in Break panel, Manual Request Editor), so the de/serialisation would be implied I guess.

The add-ons linked earlier by @kingthorin allow to view and modify the contents (`viewstate` is more interesting since it de/encodes the contents, but in the end it's the same workflow for all of them).

**Comment by su-ja** [comment]
HI,
I am new to this group and would like to know whether I could use ZAP for testing Java Deserialization issue.
Also, I am interested in contributing some work/time to this project.

**Comment by kingthorin** [comment]
@su-ja we don't yet have an active scan rule to test for deserialization issues.

**Comment by thc202** [comment]
@JStyle21 how are things progressing? Any tasks that can be split for @su-ja?

**Comment by JStyle21** [comment]
It's possible but first, @su-ja Are looking for something with a GUI? you plan to manually deserialize and send request to the server?

**Comment by JStyle21** [comment]
@su-ja 
Hi,
Sorry for the late response.
At this point what i've so far is equal to nothing and i'm fairly busy in the next few weeks.
Apart from that i'm not sure our goals are aligned as i require an API solution and the GUI option is optimal.

Therefore i suggest you either start on your own or lead this ticket and if needed we will separate into 2 threads.
